Framework/Algorithms/src/PDCalibration.cpp

```diff
@@ -495,16 +518,15 @@ void PDCalibration::exec() {
   }
   createInformationWorkspaces();
 
-  // Initialize the mask workspace, will all detectors masked by default
-  std::string maskWSName = getPropertyValue("OutputCalibrationTable");
-  maskWSName += "_mask";
-  declareProperty(std::make_unique<WorkspaceProperty<>>("MaskWorkspace", maskWSName, Direction::Output),
-                  "An output workspace containing the mask");
+  // Use the incoming mask workspace, or start with a new one if the workspace does not exist.
+  MaskWorkspace_sptr maskWS = getProperty("MaskWorkspace");
```

Comment:
I think this is where some of the problems are coming from. I think if the name isn't provided you need to create a new maskworkspace and set its name and stop messing around with the properties. At the end of the exec, the property should be set so the ADS gets its information. The current code changes the value of an input which is very unusual. Without the aid of an IDE
```c++
MaskWorkspace_sptr maskWS;
if (isDefault("MaskWorkspace")) {
  maskWS = std::make_shared<MaskWorkspace>(m_uncalibratedWS->getInstrument());
} else {
  maskWS = getWorkspace("MaskWorkspace"); // get pointer to existing workspace
}
```
later on in `exec()` there would be a call like
```c++
if (isDefault("MaskWorkspace"))
  setPropertyValue("MaskWorkspace", getPropertyValue("OutputCalibrationTable") + "_mask"); // set default name
setProperty("MaskWorkspace", maskWS->move());
```
`validateInputs()` would need changes as well.

This is mostly moving around the changes you've made because I'd never look in `validateInputs()` for a default workspace name to be created.